prime/src/components/borrow/BorrowSelectActionModal.tsx

```diff
@@ -117,6 +120,27 @@ export type BorrowSelectActionModalProps = {
 
 export default function BorrowSelectActionModal(props: BorrowSelectActionModalProps) {
   const { isOpen, setIsOpen, handleAddAction, handleAddActions } = props;
+  const [localTemplates, setLocalTemplates] = useState<ActionTemplate[]>([]);
+
+  useEffect(() => {
+    const updateLocalTemplates = () => {
+      const localTemplates = retrieveAllTemplates();
+      setLocalTemplates(localTemplates);
+    };
+    updateLocalTemplates();
+    window.addEventListener(TEMPLATE_STORED_EVENT_STRING, () => {
+      updateLocalTemplates();
+    });
+    return () => {
+      window.removeEventListener(TEMPLATE_STORED_EVENT_STRING, () => {
+        updateLocalTemplates();
+      });
```

Comment:
```suggestion
      window.removeEventListener(TEMPLATE_STORED_EVENT_STRING, updateLocalTemplates);
```

This way the add/remove listener calls are referencing the same function, instead of a unique arrow function.